table.go

```diff
@@ -69,23 +73,44 @@ func (rt *RoutingTable) Update(p peer.ID) (evicted peer.ID, err error) {
 		bucketID = len(rt.Buckets) - 1
 	}
 
+	var full = 0
+	var nonEmpty = 0
+	for i, buck := range rt.Buckets {
```

Comment:
Ideally, we could get a snapshot of all the buckets straight away, but it is really expensive. I think a better way to approach this would be to build up the view of all buckets one bucket at a time. So whichever bucket is chosen for the `Update` operation is the one we record metrics for.

Reply:
Good idea - the measures for the # of full and non-empty buckets are redundant with the utilization measure anyway. I'll rewrite this to just record utilization for the buckets we actually visit in the Update method and remove the loop.